addons/website/tools.py

```diff
@@ -187,3 +191,149 @@ def get_base_domain(url, strip_www=False):
     if strip_www and url.startswith('www.'):
         url = url[4:]
     return url
+
+# Generated primary snippet templates
+
+def generate_primary_snippet_templates(env, module):
+    refs = {}
+
+    def ref(xmlid):
+        if xmlid not in refs:
+            refs[xmlid] = env.ref(xmlid).id
+        return refs[xmlid]
+
+    def split_key(snippet_key):
+        return snippet_key.split('.') if '.' in snippet_key else ('website', snippet_key)
+
+    def create_model_data(ids, values_list):
+        env['ir.model.data'].create([{
+            'name': values_list[index]['key'].split('.')[1],
+            'module': values_list[index]['key'].split('.')[0],
+            'model': 'ir.ui.view',
+            'res_id': ids[index],
+        } for index in range(0, len(ids))])
+        # Prevent deletion by _process_end
+        for values in values_list:
+            env['ir.model.data']._load_xmlid(values["key"])
+
+    def create_missing(values_list):
+        values_list = [values for values in values_list if values]
```

Comment:
If a missing parent is somehow referenced, instead of globally crashing we survive by returning `None` from `get_create_values`.

Reply:
When does this happen? Do you have a flow that would lead to this?
Is that when writing something wrong in manifests or something?
Can't we just let it crash? As if you type a wrong xml_id anywhere in xml data, it will crash, no?

Reply:
I have not been able to identify a clear scenario, but I fear that installing/uninstalling themes might lead to some precedence issues which could be very difficult to recover from.

Reply:
Since there is no way to uninstall a theme, that should be ok no ?
But alright, let's keep it, it doesn't hurt, just comment it "Safety net in case something goes wrong, it shouldn't be possible but it's a low effort to do it" something like that
Without a comment, when I see `if values`, I am trying to figure when is this possible and trying to figure what I missed or can't understand